RecoParticleFlow/PFClusterProducerAlpaka/plugins/alpaka/PFClusterProducerKernel.dev.cc

```diff
@@ -0,0 +1,2222 @@
+#include <alpaka/alpaka.hpp>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+#include "RecoParticleFlow/PFClusterProducerAlpaka/interface/alpaka/PFClusterProducerKernel.h"
+#include "RecoParticleFlow/PFClusterProducerAlpaka/plugins/AlpakaPFCommon.h"
+
+// The following comment block is required in using the ECL-CC algorithm for topological clustering
+
+/*
+  ECL-CC code: ECL-CC is a connected components graph algorithm. The CUDA
+  implementation thereof is quite fast. It operates on graphs stored in
+  binary CSR format.
+
+  Copyright (c) 2017-2020, Texas State University. All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+
+     * Redistributions of source code must retain the above copyright
+       notice, this list of conditions and the following disclaimer.
+     * Redistributions in binary form must reproduce the above copyright
+       notice, this list of conditions and the following disclaimer in the
+       documentation and/or other materials provided with the distribution.
+     * Neither the name of Texas State University nor the names of its
+       contributors may be used to endorse or promote products derived from
+       this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL TEXAS STATE UNIVERSITY BE LIABLE FOR ANY
+  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+  Authors: Jayadharini Jaiganesh and Martin Burtscher
+
+  URL: The latest version of this code is available at
+  https://userweb.cs.txstate.edu/~burtscher/research/ECL-CC/.
+
+  Publication: This work is described in detail in the following paper.
+  Jayadharini Jaiganesh and Martin Burtscher. A High-Performance Connected
+  Components Implementation for GPUs. Proceedings of the 2018 ACM International
+  Symposium on High-Performance Parallel and Distributed Computing, pp. 92-104.
+  June 2018.
+*/
```

Comment:
Is the entire file covered by this license? If not, I'd suggest to move the parts that are into a separate file, so that the coverage of the license would be clear. Also add description what was modified from the original code.

Reply:
Moving the ECL-CC code out, would I keep the new file as a `.h` under `plugins/alpaka/`?

Reply:
It depends.

do you think the ECL-CC code could be useful in other packages ?
  - no: `plugins/alpaka/` is ok
  - yes: does it depend on alpaka ?
      - no: you could put it under `interface/`
      - yes: under `interface/alpaka/`